WordPress/Classes/Services/ReaderPostService.m

```diff
@@ -358,6 +358,12 @@ - (void)toggleFollowingForPost:(ReaderPost *)post success:(void (^)(void))succes
 
     // Define success block
     void (^successBlock)(void) = ^void() {
+        
+        // Update subscription count
+        NSInteger oldSubscriptionCount = [WPAnalytics subscriptionCount];
+        NSInteger newSubscriptionCount = follow ? oldSubscriptionCount + 1 : oldSubscriptionCount - 1;
+        [WPAnalytics setSubscriptionCount:newSubscriptionCount];
+        
```

Comment:
Updating the subscription count here as well to ensure an accurate count.

Took into account the condition below -- I realized I needed to update the subscription count if the topic was not a `ReaderSiteTopic`. 

https://github.com/wordpress-mobile/WordPress-iOS/blob/9a9e683ae5e87701701e14e1bfc4259b42cbbd67/WordPress/Classes/Services/ReaderPostService.m#L339-L344